### PR TITLE
Transclusion Enhancments

### PR DESCRIPTION
**Pull request by uroybd**
This PR implements 2 things:
  1. Transcluded block signature to id.
  2. Link in embedded section for header and block transclusion.

### CODE REVIEW

File: src/Publisher.ts [hunk 1]
```diff
@@ -371,7 +385,7 @@ export default class Publisher {
         if (currentDepth >= 4) {
             return text;
         }
-
+		const published_files = await this.getFilesMarkedForPublishing();
```
Comment (oleeskild) [inline src/Publisher.ts:388]: Can this be changed to camelCase to adhere to the convention in the project?
```suggestion
		const publishedFiles = await this.getFilesMarkedForPublishing();
```

File: src/Publisher.ts [hunk 2]
```diff
@@ -435,8 +450,11 @@ export default class Publisher {
                         const header = this.generateTransclusionHeader(headerName, linkedFile);
 
                         const headerSection = header ? `$<div class="markdown-embed-title">\n\n${header}\n\n</div>\n` : '';
-
-                        fileText = `\n<div class="transclusion internal-embed is-loaded"><div class="markdown-embed">\n\n${headerSection}\n\n`
+						let embedded_link = "";
+						if (published_files.find((f) => f.path == linkedFile.path)) {
```
Comment (oleeskild) [inline src/Publisher.ts:454]:
```suggestion
						if (publishedFiles.find((f) => f.path == linkedFile.path)) {
```

**Review by oleeskild — CHANGES_REQUESTED** [review]

**Comment by uroybd** [comment]
Done the changes based on the review. :smile:

File: src/Publisher.ts [hunk 3]
```diff
@@ -117,15 +118,29 @@ export default class Publisher {
...
-        text = await this.convertFrontMatter(text, file.path);
+		text = await this.convertFrontMatter(text, file.path);
+		text = await this.createBlockIDs(text);
         text = await this.createTranscludedText(text, file.path, 0);
         text = await this.convertDataViews(text, file.path);
         text = await this.convertLinksToFullPath(text, file.path);
         text = await this.removeObsidianComments(text);
         text = await this.createSvgEmbeds(text, file.path);
         text = await this.createBase64Images(text, file.path);
+        console.log(text);
```
Comment (oleeskild) [inline src/Publisher.ts:129]:
```suggestion
```

**Review by oleeskild — COMMENTED** [review]
Awesome! One last thing: remove the console.log statement. After that, I will merge it 😄

**Review by oleeskild — APPROVED** [review]

**Pull request merged**